### PR TITLE
Display build record validation errors

### PR DESCRIPTION
**Pull request by adammw**
Display build record validation errors

Also fix an issue where builds with periods in the folder name would not work due to failed regex match, e.g. `	
gcr.io/docker-images-180022/github.com/zendesk/consul:b04f39140a9aaf9685462584f5041b4e64f210b9`

/cc @zendesk/samson 

### Tasks
 - [x] :+1: from team

### References
 - Jira link:

### Risks
- Level: Low

### CODE REVIEW

File: plugins/gcloud/app/controllers/gcloud_controller.rb [hunk 1]
```diff
@@ -45,6 +46,5 @@ def update_build(build)
     end
 
     build.save!
```
Comment (grosser) [inline plugins/gcloud/app/controllers/gcloud_controller.rb]: can we do this to not raise/rescue and stay with the current pattern ?
`return "Failed to save build #{build.errors.full_messages}" unless build.save`

File: plugins/gcloud/app/controllers/gcloud_controller.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 class GcloudController < ApplicationController
+  class GcloudCommandError < StandardError; end
```
Comment (grosser) [inline plugins/gcloud/app/controllers/gcloud_controller.rb]: no longer need this

File: plugins/gcloud/app/controllers/gcloud_controller.rb [hunk 3]
```diff
@@ -14,8 +16,8 @@ def sync_build
     build = Build.find(params[:id])
     path = [build.project, build]
 
-    if error = update_build(build)
-      redirect_to path, alert: "Failed to sync: #{error}"
+    if (error = update_build(build))
+      redirect_to path, alert: "Failed to sync: #{e}"
```
Comment (grosser) [inline plugins/gcloud/app/controllers/gcloud_controller.rb]: e -> error ?

File: plugins/gcloud/app/controllers/gcloud_controller.rb [hunk 4]
```diff
@@ -44,7 +46,7 @@ def update_build(build)
       end
     end
 
-    build.save!
+    return "Failed to save build #{build.errors.full_messages}" unless build.save
```
Comment (grosser) [inline plugins/gcloud/app/controllers/gcloud_controller.rb]: new test ? :trollface:

**Comment by adammw** [comment]
@grosser are the test failures related? 😕

**Comment by grosser** [comment]
periodical is broken on master

**Pull request merged**